### ISSUE TITLE
Reassembling fails on ResurrectionRemix for Samsung S4

### ISSUE THREAD

**Issue by pejakm** [nagalun/needle-zip#4]
It worked previously, but with the latest build it fails. I have changed system and cache partitions to F2FS.

The error:

```
<dalvikvm> UNEXPECTED TP-LEVEL EXCEPTION:
<dalvikvm> org.jf.util.ExceptionWithContext: Unsigned short value out of range: 65536
...
! ! !  Reassembling failed, do you have more than 512 MB of RAM on this device?
```

Any thoughts?

**Comment by ale5000-git** [comment]
@nagalun: My project [Tingle](https://github.com/ale5000-git/tingle) works also inside Android.
I have converted to dalvik a more recent version of smali/baksmali and I have added a workaround for the 65536 limit.
Feel free to take the code you want.

**Comment by nagalun** [comment]
Yep, I have this abandoned, sorry :(, try what ale5000-git suggests.

**Comment by pejakm** [comment]
Thanks @ale5000-git! It worked.
